File: MacTech Demo Python/MacTech Demo/MainWindowController.py
# -*- coding: utf-8 -*-
#
#  MainWindowController.py
#  MacTech Demo
#
#  Created by Pelle on 2016-11-14.
#  Copyright (c) 2016 MacTech. All rights reserved.
#

from objc import YES, NO, IBAction, IBOutlet
from Foundation import *
from AppKit import *

class MainWindowController(NSWindowController):

    spokenTextField = IBOutlet()

    @IBAction
    def speakButtonClicked_(self, sender):
        NSLog(u"Speak button clicked")
        NSLog(self.spokenTextField.stringValue())
        scriptPath = NSBundle.mainBundle().pathForResource_ofType_(u"Script", u"sh")
        if scriptPath:
            scriptProcess = NSTask.launchedTaskWithLaunchPath_arguments_(scriptPath, [self.spokenTextField.stringValue()])
            scriptProcess.waitUntilExit()
        else:
            NSLog(u"Couldn't find Script.sh")

File: MacTech Demo Swift kopia/MacTech Demo/MainWindowController.swift
//
//  MainWindowController.swift
//  MacTech Demo
//
//  Created by MacTech on 2016-11-16.
//  Copyright © 2016 MacTech. All rights reserved.
//

import Cocoa

class MainWindowController: NSWindowController {

    @IBOutlet weak var spokenTextField: NSTextField!
    
    override func windowDidLoad() {
        super.windowDidLoad()

        // Implement this method to handle any initialization after your window controller's window has been loaded from its nib file.
    }
    
    @IBAction func pressedReturn(_ sender: Any) {
        speakButtonClicked(sender)
    }
    
    @IBAction func speakButtonClicked(_ sender: Any) {
        print("Speak button clicked")
        print("\(spokenTextField.stringValue)")
        if let scriptPath = Bundle.main.path(forResource: "Script", ofType: "sh") {
            
            let scriptProcess = Process()
            scriptProcess.launchPath = scriptPath
            scriptProcess.arguments = [spokenTextField.stringValue]
            
            let stdoutPipe = Pipe()
            scriptProcess.standardOutput = stdoutPipe
            
            scriptProcess.launch()
            scriptProcess.waitUntilExit()
            
            print("Script exited with status \(scriptProcess.terminationStatus)")
            
            let stdoutData = stdoutPipe.fileHandleForReading.readDataToEndOfFile()
            let stdout = NSString(data: stdoutData, encoding: String.Encoding.utf8.rawValue)
            print("Received stdout: \(stdout)")
            
        } else {
            print("Couldn't find Script.sh")
        }
    }
}
